test/Chainweb/Test/Rosetta/RestAPI.hs

```diff
@@ -222,65 +252,337 @@ blockTests tio envIo = testCaseSchSteps "Block Tests" $ \step -> do
+          constructValidTx ops cid' "sender00" xchain getKeys res cenv step
+
+    step "--- TRANSFER TO AN EXISTING ACCOUNT ---"
+    void $ do
+      let amt = 2.0
+          fromAcct = "sender00"
+          fromGuard = ks sender00ks
+          toAcct = "sender01"
+          toGuard = ks sender01ks
+          srcChainId = cid
+          --srcChainId = unsafeChainId 4
+          -- TOOD: non-zero chainId fails with "PollingFailure "polling check failed: {}"
+          ops = [ mkOp fromAcct (negate amt) fromGuard 0
+                , mkOp toAcct amt toGuard 1 ]
+          res = P.PLiteral $ P.LString "Write succeeded"
+      constructValidTx' ops srcChainId Nothing res
+
+    step "--- TRANSFER TO A NEW ACCOUNT ---"
+    void $ do
+      let amt = 2.0
+          fromAcct = "sender01"
+          fromGuard = ks sender01ks
+          toAcct = "someNewAccount"
+          toGuard = ks sender00ks
+          ops = [ mkOp fromAcct (negate amt) fromGuard 0
+                , mkOp toAcct amt toGuard 1 ]
+          res = P.PLiteral $ P.LString "Write succeeded"
+      constructValidTx' ops cid Nothing res
+
+    step "--- CREATE A NEW ACCOUNT ---"
+    void $ do
+      let ops = [ mkOp "anotherNewAccount" 0.0 (ks sender01ks) 0 ]
+          res = P.PLiteral $ P.LString "Write succeeded"
+      constructValidTx' ops cid Nothing res
+
+    step "--- START CROSSCHAIN TRANSFER TO NEW ACCOUNT ---"
+    let srcChain = unsafeChainId 0
+        targetChain = unsafeChainId 1
+        amt = 2.0
+        fromAcct = "sender01"
+        fromGuard = ks sender01ks
+        toAcct = "yetAnotherNewAccount"
+        toGuard = ks sender00ks
+        ops = [ mkOp fromAcct (negate amt) fromGuard 0 ]
+        targetChainText = P.ChainId $ chainIdToText targetChain
+        xchain = Just $ StartCrossChainTx
+          { _startCrossChainTx_to = toAcct
+          , _startCrossChainTx_toGuard = toGuard
+          , _startCrossChainTx_targetChain = targetChainText
+          }
+        res = P.PObject $ P.ObjectMap $ M.fromList
+              [ ("amount", (P.PLiteral $ P.LDecimal amt))
+              , ("receiver", (P.PLiteral $ P.LString toAcct))
+              , ("receiver-guard", (P.PGuard $ P.GKeySet toGuard)) ]
+    pactId <- constructValidTx' ops srcChain xchain res
+
+    step "--- FINISH CROSSCHAIN TRANSFER TO A NEW ACCOUNT ---"
+    step "get spv proof"
+    let spvReq = SpvRequest pactId targetChainText
+    TransactionOutputProofB64 spvProof <- spv srcChain cenv spvReq
+
+    let _xchainEndOps = [ mkOp toAcct amt toGuard 0 ]
+        _xchainEnd = Just $ FinishCrossChainTx
+          { _finishCrossChainTx_sourceChain = P.ChainId $ chainIdToText srcChain
+          , _finishCrossChainTx_pactId = P.toPactId $ P.unRequestKey pactId
+          , _finishCrossChainTx_proof = spvProof
+          }
+        _xchainEndRes = P.PLiteral $ P.LString "Write succeeded"
+    pure ()
+    --void $ constructValidTx' _xchainEndOps targetChain _xchainEnd _xchainEndRes
```

Comment:
@emilypi do you know what could be happening that transactions don't seem to be submitted (seem to be just dropped) when the chainId is not zero? 
I've tried using the `sending` endpoint instead of the Construction API's `submit` endpoint, but I get the same result with both.
 I've even tried simulating the `MVar` logic you implemented in `mkTransfer`, but that doesn't seem to fix it either.